src/plugins/explore/public/components/query_panel/results_summary/results_summary.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useState, useCallback, useMemo, useEffect, useRef } from 'react';
+import { useSelector } from 'react-redux';
+import { i18n } from '@osd/i18n';
+import { isEmpty } from 'lodash';
+
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
+import { EditorMode } from '../../../application/utils/state_management/types';
+import { ExploreServices } from '../../../types';
+import { ResultsSummaryButton, FeedbackStatus } from './results_summary_button';
+import { RootState } from '../../../application/utils/state_management/store';
+import { useMetrics } from './use_metrics';
+import { selectSummaryAgentIsAvailable } from '../../../application/utils/state_management/selectors';
+import { getUsageCollector } from '../../../services/usage_collector';
+import { ResultStatus } from '../../../../../data/public';
+
+const SUMMARY_REQUEST_SAMPLE_SIZE = 10;
+
+export const ResultsSummary: React.FC = () => {
+  const { services } = useOpenSearchDashboards<ExploreServices>();
+  const { core, http } = services;
+
+  const isSummaryAgentAvailable = useSelector(selectSummaryAgentIsAvailable);
+
+  const [summary, setSummary] = useState(''); // store fetched data
+  const [loading, setLoading] = useState(true); // track loading state
+  const [feedback, setFeedback] = useState(FeedbackStatus.NONE);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [generateError, setGenerateError] = useState(false);
+
+  const usageCollection = getUsageCollector();
+  const { reportMetric, reportCountMetric } = useMetrics(usageCollection);
+
+  const {
+    queryState,
+    lastExecutedPrompt,
+    dataSetState,
+    queryResults,
+    editorMode,
+    queryStatus,
+  } = useSelector((state: RootState) => ({
+    queryState: state.query,
+    lastExecutedPrompt: state.queryEditor.lastExecutedPrompt,
+    dataSetState: state.query.dataset,
+    queryResults: state.results[state.query.query as string]?.hits?.hits,
+    editorMode: state.queryEditor.editorMode,
+    queryStatus: state.queryEditor.overallQueryStatus.status,
+  }));
```

Comment:
this is more nitpick, but I would prefer having multiple useSelectors using the memoized selectors so that state updates that does not belong to these will not cause a re-render